src/PerfView/PerfViewData.cs

```diff
@@ -5744,6 +5749,39 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                                     }
                                     sumCumMetric += cumMetric;
                                     cumCount++;
+
+                                    string GetAllocationType(CallStackIndex csi)
+                                    {
+                                        if (!allocationTypeNames.TryGetValue(csi, out var typeName))
+                                        {
+                                            for (var current = csi; current != CallStackIndex.Invalid; current = eventLog.CallStacks.Caller(current))
+                                            {
+                                                var module = eventLog.CodeAddresses.ModuleFile(eventLog.CallStacks.CodeAddressIndex(current));
+                                                if (module == null)
+                                                {
+                                                    continue;
+                                                }
+
+                                                if (!loadedModules.TryGetValue(module, out var symbolModule))
+                                                {
+                                                    loadedModules[module] = symbolModule =
+                                                        (module.PdbSignature != Guid.Empty
```

Comment:
I expected there to be a direct call to get a SymbolModule from a TraceModuleFile, but I didn't find one. This is my best guess based on some other areas of the code.